geopandas/geoseries.py

```diff
@@ -272,8 +291,15 @@ def select(self, *args, **kwargs):
     def __finalize__(self, other, method=None, **kwargs):
         """ propagate metadata from other to self """
         # NOTE: backported from pandas master (upcoming v0.13)
-        for name in self._metadata:
-            object.__setattr__(self, name, getattr(other, name, None))
+        if method == "concat":
+            for name in self._metadata:
+                # using CRS of the first object
+                object.__setattr__(self, name, getattr(other.objs[0], name, None))
+            self.values.crs = other.objs[0].values.crs
```

Comment:
Also here, do you know if this is needed? Because preserving the crs of the first item is also handled in `GeometryArray._concat_same_type`, which seems to duplicate this

Reply:
I am looking into this one. And so it actually goes through `_concat_same_type`, and at that point it has the crs. But in some way it looses the crs between that point and when ending up in finalize.

Reply:
OK, so the problem lies in the `if isinstance(data, SingleBlockManager):` in the `__new__` of GeoSeries. The `self.crs = crs` on line 120 is overriding the crs of the GeometryArray stored in the SingleBlockManager. 
So we should certainly not override it (and give precedence to the crs of the values as in other places). Whether we also need to raise a warning there, I am not sure. I think we should only end up there when being called from pandas' internals, in which case crs should always be the default None. Not fully sure about that though.

Reply:
Thanks for the exploration! I have pushed a fix and consequent clean-up of `__finalize__`. I think you are right that we should end there only as a result of internal operation, so I skipped warning for that case.